### PR TITLE
net: lib: lwm2m_client_utils: fixed installing fw once it was aborted

### PR DESCRIPTION
**Pull request by andreas-chmielewski-gcx**
In case fw download is restarted a reset of dfu target is missing
causing invalid writing to dfu target.

### CODE REVIEW

**Comment by andreas-chmielewski-gcx** [comment]
As you see I am using a post_write_cb to get the state of the fw update process. Unfortunately this make it not possible for the user to use this cb in his application code. How to deal with that? Any idea?

File: subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_firmware.c [hunk 1]
```diff
@@ -173,6 +175,26 @@ static void *firmware_get_buf(uint16_t obj_inst_id, uint16_t res_id,
 	return firmware_buf;
 }
 
+static int firmware_update_state(uint16_t obj_inst_id, uint16_t res_id, uint16_t res_inst_id,
+                uint8_t *data, uint16_t data_len, bool last_block,
+                size_t total_size)
```
Comment (rlubos) [inline subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_firmware.c:180]: nit: Please align with the opening bracket.

**Review by rlubos — COMMENTED** [review]
Looks good, but there should be no merge commits in the PR, please use rebase instead to align your branch with main.

File: subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_firmware.c [hunk 2]
```diff
@@ -51,6 +51,8 @@ static int image_type = DFU_TARGET_IMAGE_TYPE_ANY;
 static char *fota_path;
 static char *fota_host;
 static int fota_sec_tag;
+static uint8_t percent_downloaded = 0;
+static uint32_t bytes_downloaded = 0;
```
Comment (VeijoPesonen) [inline subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_firmware.c:55]: No need to zero initialize static variables, _checkpatch.pl_ might even complain about it.

```
If an object that has automatic storage duration is not initialized explicitly, its value is
indeterminate. If an object that has static storage duration is not initialized explicitly,
then:
— if it has pointer type, it is initialized to a null pointer;
— if it has arithmetic type, it is initialized to (positive or unsigned) zero;
— if it is an aggregate, every member is initialized (recursively) according to these rules;
— if it is a union, the first named member is initialized (recursi vely) according to these
rules.
```
[C99 - Chapter 6.7.8](http://www.open-std.org/jtc1/sc22/wg14/www/docs/n1256.pdf)

**Comment by VeijoPesonen** [comment]
@andreas-chmielewski-gcx please rebase your work.

**Review by rlubos — APPROVED** [review]
LGTM, thanks

**Review by VeijoPesonen — APPROVED** [review]
Looks good to me too.

**Pull request merged**